.github/actions/build-with-patched-std/action.yml

```diff
@@ -0,0 +1,45 @@
+# Github composite action to build a single-source-file test binary with an
+# already-checked-out version of Rust's stdlib, that will be patched with a
+# given revision of the backtrace crate.
+
+name: Build with patched std
+description: >
+  Build a binary with a version of std that's had a specific revision of
+  backtrace patched in.
+inputs:
+  backtrace-commit:
+    description: The git commit of backtrace to patch in to std
+    required: true
+  main-rs:
+    description: The (single) source code file to compile
+    required: true
+  rustc-dir:
+    description: The root directory of the rustc repo
+    required: true
+outputs:
+  test-binary-size:
+    description: The size in bytes of the built test binary
+    value: ${{ steps.measure.outputs.test-binary-size }}
+runs:
+  using: "composite"
+  steps:
+    - shell: bash
+      id: measure
+      env:
+        RUSTC_FLAGS: -Copt-level=s -Cstrip=symbols
+        RUSTC_BUILD_DIR: build/x86_64-unknown-linux-gnu
```

Comment:
Only difference to allow this build on windows is that string?

Reply:
Yes, https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell

replace `runs-on: ubuntu-latest` with `windows-2019`.

Reply:
Oh I saw that, but in order to test it locally, is there a published Docker image that relates to? AFAIK Github make their own modifications to underlying images to support their CI eg. `ubuntu-latest` is the official `ubuntu:latest` Docker image plus various GH/CI related changes. But I have not been able to find out whether these images are available to pull, or how to refer to them if so.

Reply:
I don't think windows can run in docker, it's some azure stuff that runs windows images here. You can try to test it on your fork of backtrace, i think.

Reply:
Ah that makes so much more sense. Yes, I have a weird fork setup where I test one fake test branch against another. I'll give it a shot.

Reply:
Actually I think it can just be replaced with `build/host`.

Reply:
I got this working on Windows without a lot of changes at all, [see this CI run](https://github.com/detly/backtrace-rs/actions/runs/5595206423/jobs/10230858124). I'll port the changes over but leave the jobs as running on `ubuntu-latest` unless there's a preference for `windows-2019`.

Reply:
I don't understand. If you leave only linux runner, how it will catch changes for windows? You can specify [matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) so it will run on windows-2019 and ubuntu-22.04. Mac can be skipped, i think.

Reply:
Fair point, especially since a lot of contribution to code size comes from platform specific code (eg. gimli and coff). But the downside to this would be that for changes that affect both, you'll get two comments per PR _per update_, which might be a bit spammy. (Plus, if you ever want to add a dimension to the matrix eg. compiler flags, features, you'll potentially get N×M comments per PR update.)

I had thought there would be a way to change the comment job to summarise it all, but AFAICT Github [don't have a way](https://github.com/orgs/community/discussions/17245) to operate on the combined/collected output from matrices. Even using outputs or artifacts seems like it's [subject to race conditions](https://github.com/orgs/community/discussions/17245#discussioncomment-6261191). It's designed for matrix elements to map to independent workflows that don't need the full set of results later.

Reply:
I guess it's ok to have 2 comments instead of 1 for now, it can be tweaked later. Looks like you can edit comments, so potentially it can stack all of them for one commit into one, but that can be done later.

Reply:
I was wrong about the two comments thing. If other jobs have a `needs`-dependency on matrix jobs, they run only once, and get only the last output (with order being arbitrary). So accessing outputs is possible by uploading artifacts where collisions are avoided by having distinct (matrix-based) filenames. The comment step would then loop over these outputs (knowing that all are complete, because it gets fired on the last job).

I will see if this works and let you know.

Reply:
Not gonna lie, this is more javascript than I've written for a while. But. I wanted to make it not-horrible to add dimensions or values to the matrix in the future. Basically, the size-measuring-jobs write a file each to an artifact store, and the reporting job reads them all. If there are any changes, it posts a comment that looks like:

---

Code size changes for a hello-world Rust program linked with libstd with backtrace:

On platform `ubuntu-latest`:

- Original binary size: **399,760 B**
- Updated binary size: **379,280 B**
- Difference: **-20,480 B** (-5.12%)

On platform `windows-latest`:

- Original binary size: **160,256 B**
- Updated binary size: **151,552 B**
- Difference: **-8,704 B** (-5.43%)